libs/solid/skeleton/src/lib/components/base-layout/base-layout.component.ts

```diff
@@ -53,6 +61,14 @@ export class BaseLayoutComponent implements OnInit {
       });
   }
 
+  public ngAfterViewInit(): void {
+    setTimeout(() => {
+      this.introService.guidedTour((_targetElement: any) => {
+        return;
```

Comment:
Ich glaube (ich bin aber nicht sicher), dass wir hier alle "Actions" einfügen können.
`_targetElement.id` gibt uns hier den _nächsten_ `step` zurück.

Dann können wir für verschiedene Steps einen Befehl ausführen, z.B. `this.MainMenu?.open()` wenn `this.FixedLayout == false` ist.

Reply:
Also die "Actions" können wir hier ausführen. Wir haben nur ein kleines Problem, weil IntroJS die Elemente nach der "Actions" nicht richtig erkennen kann

Reply:
![Screenshot (53)](https://user-images.githubusercontent.com/33594579/159589559-41171954-923c-4c6f-8931-42f821b0c002.png)
Z.B hier wird das Menü nicht hervorgehoben

Reply:
Das habe ich auch gesehen. Ich habe aber das hier probiert:

```
  public ngAfterViewInit(): void {
    setTimeout(() => {
      this.introService.guidedTour((_targetElement: any) => {
        if (_targetElement.id == 'step1') {
          this.MainMenu?.open();
        }
        return;
      });
    }, 1000);
```

Stimmt, Step1 funktioniert nicht - aber das könnten wir fixen, indem [an dieser Stelle](https://github.com/zentrumnawi/solid-frontend/blob/bdfbfb10f1412ec7fe0134fa126be69ec4a696bc/libs/solid/skeleton/src/lib/components/base-layout/base-layout.component.html#L30) noch ein `id="step1"` eingefügt wird. Dann wird im mobilen Layout der Menu-Button highlighted und im FixedLayout das gesamte Menü.  

Ansonsten ist `onbeforechange()` vielleicht nicht geeignet und wir müssen einen anderen hook verwenden (`onchange()` oder `onafterchange()`).

Also - ich denke, das ist durch die richtige Kombination hooks und steps fast alles möglich ist - bei mir funktioniert das mit den beiden Änderungen oben ganz gut! (Das Menü muss aber auch am Ende wieder geschlossen werden :wink:)

Reply:
So mit dem Menu-Button habe ich auch gedacht 👍